### ISSUE TITLE
The dump method of the EnergySystem class does not work after an optimisation with solph

### ISSUE THREAD

**Issue by uvchik** [oemof/oemof#81]
If I dump the energy system before the optimisation everything works fine, but after the optimisation it does not work any more.

```python
Traceback (most recent call last):
  File "/home/uwe/rli-lokal/git_home/oemof/examples/storage_optimization/storage_invest.py", line 159, in <module>
    energysystem.dump()
  File "/home/uwe/rli-lokal/git_home/oemof/oemof/core/energy_system.py", line 154, in dump
    pickle.dump(self.__dict__, open(os.path.join(dpath, filename), 'wb'))
_pickle.PicklingError: Can't pickle <function add_storage_charge_discharge_limits.<locals>.storage_discharge_limit_rule at 0x7f6af5cedd90>: attribute lookup storage_discharge_limit_rule on oemof.solph.linear_constraints failed
```

**Comment by uvchik** [comment]
@gnn : I would like to write a test so that this will not happen again but how can I do it without using the solver?

**Comment by uvchik** [comment]
Closed, because it is a duplicate of #74.
